perfkitbenchmarker/pkb.py

```diff
@@ -195,6 +195,51 @@
 events.initialization_complete.connect(traces.RegisterAll)
 
 
+def _CreateBenchmarkRunList():
+  """Create a list of configs and states for each benchmark run to be scheduled.
+
+  Returns:
+    list of (args, run_status_list) pairs. Contains one pair per benchmark run
+    to be scheduled, including multiple pairs for benchmarks that will be run
+    multiple times. args is a tuple of the first five arguments to pass to
+    RunBenchmark, and run_status_list is a list of strings in the order of
+    [benchmark_name, benchmark_uid, benchmark_status].
+  """
+  result = []
+  benchmark_tuple_list = benchmark_sets.GetBenchmarksFromFlags()
+  total_benchmarks = len(benchmark_tuple_list)
+  benchmark_counts = collections.defaultdict(itertools.count)
+  for i, benchmark_tuple in enumerate(benchmark_tuple_list):
+    # Construct benchmark config object.
+    benchmark_module, user_config = benchmark_tuple
+    name = benchmark_module.BENCHMARK_NAME
+    config_dict = benchmark_module.GetConfig(user_config)
+    config_spec_class = getattr(
+        benchmark_module, 'BENCHMARK_CONFIG_SPEC_CLASS',
+        benchmark_config_spec.BenchmarkConfigSpec)
+    config = config_spec_class(name, flag_values=FLAGS, **config_dict)
+
+    # Assign a unique ID to each benchmark run. This differs even between two
+    # runs of the same benchmark within a single PKB run.
+    uid = name + str(benchmark_counts[name].next())
+
+    # Optional step to check flag values and verify files exist.
+    check_prereqs = getattr(benchmark_module, 'CheckPrerequisites', None)
+    if check_prereqs:
+      try:
+        with config.RedirectFlags(FLAGS):
+          check_prereqs()
```

Comment:
This isn't a change from the way it worked before, but do we want to re-check prerequisites every time we run a stage or just on the "provision" stage?